vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -1181,15 +1155,54 @@ protected Column<?> getBottomLevelColumn() {
         private void setupColumnEditor() {
             editorRenderer = new EditorRenderer<>((Editor) grid.getEditor(),
                     columnInternalId);
+            setupEditorRenderer();
+        }
+
+        private void setupRenderer() {
+            if (renderer == null) {
+                return;
+            }
+            rendering = renderer.render(getElement(), (KeyMapper<T>) getGrid()
+                    .getDataCommunicator().getKeyMapper());
+            columnDataGeneratorRegistration = rendering.getDataGenerator()
+                    .map(dataGenerator -> grid
+                            .addDataGenerator((DataGenerator) dataGenerator))
+                    .orElse(null);
+            grid.getDataProvider().refreshAll();
```

Comment:
I agree that it would be better to do it that way. However, directly calling in `setVisible` does not work and I'm not sure what would be a good way to refactor the refresh.